R/derive_merged.R

```diff
@@ -865,3 +865,85 @@ derive_var_merged_character <- function(dataset,
     mutate(!!new_var := if_else(temp_match_flag, !!new_var, missing_value, missing_value)) %>%
     select(-temp_match_flag)
 }
+
+
+#' Merge Lookup Table with Source Dataset
+#'
+#' Merge user-defined lookup table with the input dataset. Optionally print a
+#' list of records from the input dataset that do not have corresponding
+#' mapping from the lookup table.
+#'
+#' @param dataset_add Lookup table
+#'
+#' The variables specified by the `by_vars` parameter are expected.
+#'
+#' @param print_not_mapped Print a list of unique `by_vars` values that do not
+#' have corresponding records from the lookup table?
+#'
+#' *Default*: TRUE
+#'
+#' *Permitted Values*: TRUE, FALSE
+#'
+#' @inheritParams derive_vars_merged
+#'
+#'
+#' @return The output dataset contains all observations and variables of the
+#' input dataset, and add the variables specified in `new_vars from the lookup
+#' table specified in `dataset_add`. Optionally prints a list of unique
+#' `by_vars` values that do not have corresponding records
+#' from the lookup table (by specifying `print_not_mapped = TRUE`).
+#'
+#' @author Annie Yang
+#'
+#' @keywords derivation adam
+#'
+#' @export
+#'
+#' @examples
+#' library(admiral.test)
+#' library(dplyr, warn.conflicts = FALSE)
+#' data("admiral_vs")
+#' param_lookup <- tibble::tribble(
+#'   ~VSTESTCD, ~VSTEST, ~PARAMCD, ~PARAM,
+#'   "SYSBP", "Systolic Blood Pressure", "SYSBP", "Systolic Blood Pressure (mmHg)",
+#'   "WEIGHT", "Weight", "WEIGHT", "Weight (kg)",
+#'   "HEIGHT", "Height", "HEIGHT", "Height (cm)",
+#'   "TEMP", "Temperature", "TEMP", "Temperature (C)",
+#'   "MAP", "Mean Arterial Pressure", "MAP", "Mean Arterial Pressure (mmHg)",
+#'   "BMI", "Body Mass Index", "BMI", "Body Mass Index(kg/m^2)",
+#'   "BSA", "Body Surface Area", "BSA", "Body Surface Area(m^2)"
+#' )
+#' derive_vars_merged_lookup(
+#'   dataset = admiral_vs,
+#'   dataset_add = param_lookup,
+#'   by_vars = vars(VSTESTCD),
+#'   new_vars = vars(PARAMCD),
+#'   print_not_mapped = TRUE
+#' )
+derive_vars_merged_lookup <- function(dataset,
+                                      dataset_add,
+                                      by_vars,
+                                      new_vars = NULL,
+                                      print_not_mapped = TRUE) {
+  assert_logical_scalar(print_not_mapped)
+  res <- derive_vars_merged(
+    dataset,
+    dataset_add,
+    by_vars = by_vars,
+    new_vars = new_vars,
+    match_flag = temp_match_flag
+  )
+
+  if (print_not_mapped) {
+    temp_not_mapped <- res %>%
+      filter(is.na(temp_match_flag)) %>%
+      distinct(!!!by_vars)
+
```

Comment:
Could we also provide the complete dataset to the user? Printing just shows the first ten rows but you could have much more.

Reply:
Thanks for the suggestion. I updated to follow a similar approach as `signal_duplicate_records()` to allow user to use `get_not_mapped()` to obtain the complete dataset. Please let me know if an alternative approach is preferred.